### DISCUSSION TITLE
Camera with Stabilizer

### DISCUSSION THREAD

**Discussion by damavand1** [ultralytics/ultralytics#9257] [General]
Hi

I'm try to build delivery robot, like coco delivery and starship delivery

now, at least i need 2 camera for front and rear view

camera should be wide enough, and very very good **stabilizer**, fast connection , and ubuntu known  it as camera (web cam like)
I tested go pro 12, is very good, but some problem it have
1- expensive Price
2- don't connect to ubuntu officially 
3- battery needed  (robot have main battery)
4- some unusable line 2 lcd, that makes it expensive
and ...  

small robots have very vibration in moving time
i needs a camera that gives me somethings like this [this](https://www.youtube.com/watch?v=nUOS4N4cD_Q)

now, what's your suggestion for camera ?

special thanks

**Comment by glenn-jocher** [comment]
@damavand1 hello!

For a delivery robot that requires reliable front and rear vision, having cameras with good stabilization and compatibility with Ubuntu is key. Considering your concerns with GoPro 12 such as price and connectivity, you might want to explore other action cameras that are marketed as having great image stabilization and are more budget-friendly. Brands like DJI, with their Osmo Action, or the AKASO series, might offer what you're looking for in terms of stabilization, wide-angle views, fast connection speeds, and better compatibility with external power sources. 

For Ubuntu compatibility, many cameras support USB video class (UVC) for plug-and-play functionality. You can check this using:
```bash
lsusb
```
to identify the camera and then use tools like `ffmpeg` or `guvcview` to capture the feed.

Remember, external battery packs or direct power supply conversions for your robot can tackle power challenges. It's a bit of trial and experimentation to find the perfect fit, but there are plenty of options out there. Good luck with your delivery robot! 🤖🚀

Best wishes.
